lib/rules/no-actions-hash.js

```diff
@@ -0,0 +1,49 @@
+const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+const ERROR_MESSAGE = 'Use the @action decorator instead of declaring an actions hash';
+
+module.exports = {
+  ERROR_MESSAGE,
+  meta: {
+    docs: {
+      description: 'Disallows the actions hash in components, controllers and routes',
+      category: 'Ember Octane',
+      recommended: false,
+    },
+    fixable: null, // or "code" or "whitespace"
+    url:
+      'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-actions-hash.md',
+  },
+
+  create: context => {
+    function inClassWhichCanContainActions(context, node) {
+      return (
+        ember.isEmberComponent(context, node) ||
+        ember.isEmberController(context, node) ||
+        ember.isEmberRoute(context, node)
+      );
+    }
+
+    function reportActionsProp(properties) {
+      properties.forEach(property => {
```

Comment:
Let's switch to `find` instead of `forEach` since we know there's only one `actions` property we need to find.

Reply:
Nicer to read as well. Done!